engines/alg/game_bountyhunter.cpp

```diff
@@ -0,0 +1,1343 @@
+			_vm->quitGame();
+		}
+	}
+	return Common::kNoError;
+}
+
+void GameBountyHunter::_NewGame() {
+	_playerLives[0] = _playerLives[1] = 3;
+	_playerShots[0] = _playerShots[1] = 6;
+	_playerGun[0] = _playerGun[1] = 1;
+	_playerScore[0] = _playerScore[1] = 0;
+	_currentSubLevelSceneId = 0x017B;
+	_holster = false;
+}
+
+void GameBountyHunter::_DoMenu() {
+	uint32 startTime = _GetMsTime();
+	_RestoreCursor();
+	_DoCursor();
+	_inMenu = true;
+	_MoveMouse();
+	g_system->getMixer()->pauseHandle(_sceneAudioHandle, true);
+	_screen->copyRectToSurface(_background->getBasePtr(_videoPosX, _videoPosY), _background->pitch, _videoPosX, _videoPosY, _videoDecoder->getWidth(), _videoDecoder->getHeight());
+	while (_inMenu && !_vm->shouldQuit()) {
+		Common::Point firedCoords;
+		if (__Fired(&firedCoords)) {
+			Rect *hitMenuRect = _CheckZone(_submenzone, &firedCoords);
+			if (hitMenuRect != nullptr) {
+				callScriptFunctionRectHit(hitMenuRect->rectHit, hitMenuRect);
+			}
+		}
+		g_system->delayMillis(15);
+		updateScreen();
+	}
+	_RestoreCursor();
+	_DoCursor();
+	g_system->getMixer()->pauseHandle(_sceneAudioHandle, false);
+	if (_hadPause) {
+		unsigned long endTime = _GetMsTime();
+		unsigned long timeDiff = endTime - startTime;
+		_pauseTime += timeDiff;
+		_nextFrameTime += timeDiff;
+	}
+}
+
+void GameBountyHunter::_DoCursor() {
+	_UpdateMouse();
+}
+
+void GameBountyHunter::_UpdateMouse() {
+	if (_oldWhichGun != _whichGun) {
+		Graphics::PixelFormat pixelFormat = Graphics::PixelFormat::createFormatCLUT8();
+		Graphics::Surface *cursor = &(*_gun)[_whichGun];
+		if (_playerGun[0] == 2 && _whichGun < 2) {
+			cursor = &(*_shotgun)[_whichGun];
+		}
+		CursorMan.popAllCursors();
+		uint16 hotspotX = (cursor->w / 2) + 8;
+		uint16 hotspotY = (cursor->h / 2) + 8;
+		if (debugChannelSet(1, Alg::kAlgDebugGraphics)) {
+			cursor->drawLine(0, hotspotY, cursor->w, hotspotY, 1);
+			cursor->drawLine(hotspotX, 0, hotspotX, cursor->h, 1);
+		}
+		CursorMan.pushCursor(cursor->getPixels(), cursor->w, cursor->h, hotspotX, hotspotY, 0, false, &pixelFormat);
+		CursorMan.showMouse(true);
+		_oldWhichGun = _whichGun;
+	}
+}
+
+void GameBountyHunter::_MoveMouse() {
+	if (_inMenu) {
+		_whichGun = 3; // in menu cursor
+	} else {
+		// disabled for now, because glitchy
+		/*
+		uint16 x = _mousePos.x;
+		uint16 y = _mousePos.y;
+		if (x < 13) {x = 13;}
+		if (x > 286) {x = 286;}
+		if (y < 3) {y = 3;}
+		if (y > 166) {y = 166;}
+		if (_mousePos.x != x || _mousePos.y != y) {
+			_mousePos.x = x;
+			_mousePos.y = y;
+			g_system->warpMouse(x, y);
+		}
+		*/
+		if (_mousePos.y >= 0xA3 && _mousePos.x >= 0xF0) {
+			_whichGun = 1; // holster
+		} else if (_mousePos.y >= 0xA3 && _mousePos.x <= 0x43) {
+			_whichGun = 2; // menu button cursor
+		} else {
+			_whichGun = 0; // regular gun
+		}
+	}
+	_UpdateMouse();
+}
+
+void GameBountyHunter::_DisplayLivesLeft(uint8 player) {
+	if (_lives == _oldLives) {
+		return;
+	}
+	int posY = 0x67;
+	for (uint8 i = 0; i < 3; i++) {
```

Comment:
```suggestion
	for (int i = 0; i < 3; i++) {
```
This will perform better (not that it matters on 3 steps), but also gives better code readability

Reply:
fixed